sycl/include/sycl/detail/pi.h

```diff
@@ -582,6 +588,8 @@ constexpr pi_usm_mem_properties PI_MEM_USM_ALLOC_BUFFER_LOCATION = 0x419E;
 // NOTE: queue properties are implemented this way to better support bit
 // manipulations
 using pi_queue_properties = pi_bitfield;
+constexpr pi_queue_properties PI_QUEUE_FLAGS = -1;
+constexpr pi_queue_properties PI_QUEUE_COMPUTE_INDEX = -2;
```

Comment:
What does this -1/-2 achieve? Also please document it so it is clear that PI_QUEUE_FLAGS keeps a bit-mask of these possible values below. Maybe rename them to PI_QUEUE_FLAG_... for clarity?

Reply:
> What does ... achieve

I wanted to have the value that wouldn't be possible to be created using bit-or of the flags below, including potentially new flags introduced in future. That meant that the leftmost bit needs to be set, and then using negative numbers was just the easiest way to use (so that I wouldn't have to care about the bitwidth).

> Maybe rename them to PI_QUEUE_FLAG_... for clarity

That would be unnecessary backward-incompatible change, IMO. We could introduce aliases to preserve it but I wouldn't like that either.

Reply:
> I wanted to have the value that wouldn't be possible to be created using bit-or of the flags below

OK, good.

> > Maybe rename them to PI_QUEUE_FLAG_... for clarity
> 
> That would be unnecessary backward-incompatible change, IMO. We could introduce aliases to preserve it but I wouldn't like that either.

They are not user-visible, we can name them whatever